make/os_mac

```diff
@@ -5,7 +5,6 @@
 #   - GTEST_CXXFLAGS
 
 ifeq (g++,$(CC_TYPE))
-  CXXFLAGS += -Wno-unused-function
```

Comment:
did you intend to remove this line?

Reply:
Woops, yes but maybe it should have been a separate PR - I meant to take it out in the last PR where I moved this warning and another into the default options